From: David Bauer <mail@david-bauer.net>
Date: Mon, 29 Jun 2020 12:24:04 +0200
Subject: package: add ravpower-mcu package

This package allows to read battery status information and control the
power state of the RAVPower RP-WD009 power management IC.

Signed-off-by: David Bauer <mail@david-bauer.net>
(cherry picked from commit 1ba0466d43efd03621eb491a4c52b5ae16f2ace5)
(cherry picked from commit b09516ae6abea3976a367c38d66bc26e73096e7b)

diff --git a/package/utils/ravpower-mcu/Makefile b/package/utils/ravpower-mcu/Makefile
new file mode 100644
index 0000000000000000000000000000000000000000..435445fcc5d03459ab8291e87b17c1b679df8705
--- /dev/null
+++ b/package/utils/ravpower-mcu/Makefile
@@ -0,0 +1,34 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=ravpower-mcu
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL=https://github.com/blocktrron/ravpower-mcu.git
+PKG_MIRROR_HASH:=edde0fda3fc708eac65baff46b5b6f1290ab733d3f4bd7fc027ef45c3d1b5814
+PKG_SOURCE_DATE:=2020-06-19
+PKG_SOURCE_VERSION:=1665d9e9212dcd118629a74fbe658841f81036f7
+PKG_MAINTAINER:=David Bauer <mail@david-bauer.net>
+PKG_LICENSE:=GPL-2.0-or-later
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/ravpower-mcu
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Utility to control the RAVPower RP-WD009 PMIC
+  DEPENDS:=@TARGET_ramips_mt76x8
+  URL:=https://github.com/blocktrron/ravpower-mcu/
+endef
+
+define Build/Compile
+	$(MAKE) -C $(PKG_BUILD_DIR) \
+		CC="$(TARGET_CC)" \
+		CFLAGS="$(TARGET_CFLAGS) -Wall"
+endef
+
+define Package/ravpower-mcu/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/out/ravpower-mcu $(1)/usr/bin/
+endef
+
+$(eval $(call BuildPackage,ravpower-mcu))
diff --git a/target/linux/ramips/image/mt76x8.mk b/target/linux/ramips/image/mt76x8.mk
index 97802d2668323ad54d2bb3d7b4f2812911eb6f6e..3f46b9ae85fb90a5aa9daa4c69f1dfbc10321bb6 100644
--- a/target/linux/ramips/image/mt76x8.mk
+++ b/target/linux/ramips/image/mt76x8.mk
@@ -179,7 +179,8 @@ define Device/ravpower_rp-wd009
   DTS := RP-WD009
   IMAGE_SIZE := 14272k
   DEVICE_TITLE := RAVPower RP-WD009
-  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620 kmod-i2c-mt7628
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb2 kmod-usb-ohci \
+	kmod-sdhci-mt7620 kmod-i2c-mt7628 ravpower-mcu
 endef
 TARGET_DEVICES += ravpower_rp-wd009
 
